### PR TITLE
Add rbd disk support to set_vm_disk.

### PR DESCRIPTION
**Pull request by rbian**

### CODE REVIEW

**Comment by waynesun09** [comment]
ACK

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -2048,6 +2052,20 @@ def set_vm_disk(vm, params, tmp_dir=None):
...
+    elif disk_src_protocol == 'rbd':
+        mon_host = params.get("mon_host")
+        if image_convert:
+            disk_cmd = ("qemu-img convert -f %s -O %s %s rbd:%s:mon_host=%s"
+                        % (src_disk_format, disk_format, blk_source,
+                           disk_src_name, mon_host))
+        else:
+            disk_cmd = ("qemu-img create -f %s rbd:%s:mon_host=%s 10M" %
+                        (disk_format, disk_src_name, mon_host))
+        utils.run(disk_cmd, ignore_status=False)
```
Comment (PandaWei) [inline virttest/utils_test/libvirt.py]: No need to set ignore_status to False explicitly.
The default is False and a lot of calls for utils.run doesn't set False.
Forgive me for carp. ^_^

Others good.  ACK

**Comment by rbian** [comment]
Depends on #2222

**Comment by waynesun09** [comment]
@rbian pls rebase this PR to resolve conflict

**Comment by waynesun09** [comment]
@rbian thanks, merge now.

**Pull request merged**